02_C# Fundamentals/SystemWatcherApp/SystemWatcherApp/Program.cs

```diff
@@ -0,0 +1,147 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using SystemWatcherApp.Configs;
+using SystemWatcherApp.Resources;
+
+namespace SystemWatcherApp
+{
+    class Program
+    {
+        readonly static List<string> startDirectories;
+        readonly static List<FileSystemWatcher> watchers;
+        readonly static string defaultDirectory = @"D:\_02_BCL\Default\";
+        static CultureInfo currentCulture;
+        static List<RuleElement> rules;
+
+        static Program()
+        {
+            rules = new List<RuleElement>();
+            startDirectories = new List<string>();
+            watchers = new List<FileSystemWatcher>();
+        }
+
+        static void Main(string[] args)
+        {
+            Console.OutputEncoding = Encoding.UTF8;
+
+            WatcherSectionGroup watcherSectionGroup = ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.None).SectionGroups["watcherSection"] as WatcherSectionGroup;
+
+            foreach (ConfigurationSection section in watcherSectionGroup.Sections)
```

Comment:
Now this is something very weird. Are you cycling through sections here? But don't you already have an object of a class, that has config section mapped onto it? Your `WatcherSectionGroup` even have properties for accessing `RuleSection` and `LocaleSection`, which are never used for some reason.

Reply:
I have refactored this